`ConnectionResetError` when launching GCE instances

During security issue validation recently, I saw this error multiple times across multiple runs of the cloud-init integration tests against GCE:

```pytb
==================================== ERRORS ====================================
_ ERROR at setup of TestPackageUpdateUpgradeInstall.test_new_packages_are_installed _

request = <SubRequest 'class_client' for <Function test_new_packages_are_installed>>
fixture_utils = <class 'conftest._FixtureUtils'>
session_cloud = <tests.integration_tests.clouds.GceCloud object at 0x7fe4c52650a0>
setup_image = None

    @pytest.yield_fixture(scope='class')
    def class_client(request, fixture_utils, session_cloud, setup_image):
        """Provide a client that runs once per class."""
>       with _client(request, fixture_utils, session_cloud) as client:

tests/integration_tests/conftest.py:236: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
tests/integration_tests/conftest.py:210: in _client
    with session_cloud.launch(
tests/integration_tests/clouds.py:165: in launch
    pycloudlib_instance = self._perform_launch(kwargs)
tests/integration_tests/clouds.py:135: in _perform_launch
    pycloudlib_instance = self.cloud_instance.launch(**launch_kwargs)
../../dev/pycloudlib/pycloudlib/gce/cloud.py:213: in launch
    self._wait_for_operation(operation, operation_type='zone')
../../dev/pycloudlib/pycloudlib/gce/cloud.py:300: in _wait_for_operation
    response = api.get(**kwargs).execute()
../../.virtualenvs/cloud-init/lib/python3.8/site-packages/googleapiclient/_helpers.py:130: in positional_wrapper
    return wrapped(*args, **kwargs)
../../.virtualenvs/cloud-init/lib/python3.8/site-packages/googleapiclient/http.py:844: in execute
    resp, content = _retry_request(
../../.virtualenvs/cloud-init/lib/python3.8/site-packages/googleapiclient/http.py:164: in _retry_request
    resp, content = http.request(uri, method, *args, **kwargs)
../../.virtualenvs/cloud-init/lib/python3.8/site-packages/google_auth_httplib2.py:200: in request
    response, content = self.http.request(
../../.virtualenvs/cloud-init/lib/python3.8/site-packages/httplib2/__init__.py:1985: in request
    (response, content) = self._request(
../../.virtualenvs/cloud-init/lib/python3.8/site-packages/httplib2/__init__.py:1650: in _request
    (response, content) = self._conn_request(
../../.virtualenvs/cloud-init/lib/python3.8/site-packages/httplib2/__init__.py:1589: in _conn_request
    response = conn.getresponse()
/usr/lib/python3.8/http/client.py:1347: in getresponse
    response.begin()
/usr/lib/python3.8/http/client.py:307: in begin
    version, status, reason = self._read_status()
/usr/lib/python3.8/http/client.py:268: in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
/usr/lib/python3.8/socket.py:669: in readinto
    return self._sock.recv_into(b)
/usr/lib/python3.8/ssl.py:1241: in recv_into
    return self.read(nbytes, buffer)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ssl.SSLSocket [closed] fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6>
len = 8192, buffer = <memory at 0x7fe4c414b280>

    def read(self, len=1024, buffer=None):
        """Read up to LEN bytes and return them.
        Return zero-length string on EOF."""
    
        self._checkClosed()
        if self._sslobj is None:
            raise ValueError("Read on closed or unwrapped SSL socket.")
        try:
            if buffer is not None:
>               return self._sslobj.read(len, buffer)
E               ConnectionResetError: [Errno 104] Connection reset by peer
```

As this seems to be a common error condition, pycloudlib should probably handle it for consumers.

--- a/pycloudlib/gce/cloud.py
+++ b/pycloudlib/gce/cloud.py
@@ -307,9 +307,17 @@ class GCE(BaseCloud):
         else:
             api = self.compute.globalOperations()
         for _ in range(sleep_seconds):
-            response = api.get(**kwargs).execute()
-            if response['status'] == 'DONE':
-                break
+            try:
+                response = api.get(**kwargs).execute()
+            except ConnectionResetError:
+                # This exception is known to be raised by GCE every so often:
+                # https://github.com/canonical/pycloudlib/issues/101.
+                response = {
+                    "status": "ConnectionResetError", "statusMessage": "n/a",
+                }
+            else:
+                if response['status'] == 'DONE':
+                    break
             time.sleep(1)
         else:
             raise Exception(

